### PR TITLE
Add Scan and UnmarshalWKB methods

### PR DESCRIPTION
**Pull request by paulmach**
Scan implements the [sql.Scanner](http://golang.org/pkg/database/sql/#Scanner) interface allowing points, lines, point sets and paths to be passed into rows.Scan(…interface{}) calls. This simplifies working with the database a long.

Basically allows you to do stuff like
```
row := db.QueryRow("SELECT AsBinary(point_column) FROM table")

var p *geo.Point
row.Scan(&p)
```

Note that you need to wrap the column in `AsBinary`

@mlerner

### CODE REVIEW

**Comment by paulmach** [comment]
I'm currently debating if the `.Scan` methods are useful or confusing since you have to wrap the stuff in `AsBinary`. Also, unmarshal is pretty annoying since you have to make sure the stuff isn't nil. An alternate approach:
```
row, _ := db.Query("SELECT point_column FROM table")
row.Next()
defer row.Close()

var raw sql.RawData
row.Scan(&raw)
p, _ := geo.NewPointFromWKB(raw[4:])
```

**Comment by paulmach** [comment]
Updated as discussed in the previous comment. There are now `NewPointFromWKB`, `NewLineFromWKB`, etc. methods.

Scan now supports WKB or mysqls SRID+WKB encoding. See the updates to the readme for more details.

File: wkb.go [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+var (
+	// ErrUnsupportedDataType is returned by Scan methods when asked to scan
+	// non []byte data from the database. This should never happen
+	// if the driver is acting appropriately.
+	ErrUnsupportedDataType = errors.New("go.geo: scan value must be []byte")
+
+	// ErrNotWKB is returned when unmarshalling WKB and the data is not valid.
+	ErrNotWKB = errors.New("go.geo: invalid WKB data")
+
+	// ErrIncorrectGeometry is returned when unmarshalling WKG data into the wrong type.
```
Comment (mlerner) [inline wkb.go]: `returned when unmarshalling WKB data into`?

File: wkb.go [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+	// For example, unmarshaling linestring data into a point.
+	ErrIncorrectGeometry = errors.New("go.geo: incorrect geometry")
+)
+
+// NewPointFromWKB will take raw WKB and set the data for a new point.
+// The WKB data must of type Point. Will return nil if invalid WKB point.
```
Comment (mlerner) [inline wkb.go]: 'The WKB data must be of type Point'

File: wkb.go [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+	if data[0] == 0 || data[0] == 1 {
+		if err := ps.unmarshalWKB(data); err == nil {
+			return nil
+		}
+	}
+
+	return ps.unmarshalWKB(data[4:])
+}
+
+func (ps *PointSet) unmarshalWKB(data []byte) error {
```
Comment (mlerner) [inline wkb.go]: What about making an interface that PointSet, Point, and Path all implement and putting that interface in this file? Then  each geo object can implement the methods. That might make it easier for others to find the code in the library.

**Pull request merged**